Added graceful handling of empty arrays.

thinkingmedia opened the pull request:
Fixes some unexpected errors when trying to output a table for an empty array, or an incorrectly structured array.

thinkingmedia committed:
Added graceful handling of empty arrays.

Fixes some unexpected errors when trying to output a table for an empty array, or an incorrectly structured array.

thinkingmedia committed:
Update TableHelper.php

thinkingmedia committed:
Update TableHelper.php

chinpei215 commented on src/Shell/Helper/TableHelper.php:
```diff
@@ -78,6 +78,10 @@ protected function _rowSeparator($widths)
      */
     protected function _render($row, $widths, $options = [])
     {
+        if(empty($rows) || !is_array($row)) {
```
Typo. `empty()` is not suitable to evaluate variables. Use `!$rows` instead. See #8397

markstory replied:
Or count() if you know it is an array already.

chinpei215 commented on src/Shell/Helper/TableHelper.php:
```diff
@@ -98,6 +102,10 @@ protected function _render($row, $widths, $options = [])
      */
     public function output($rows)
     {
+        if(empty($rows) || !is_array($rows)) {
```
Not typo, but use `!$rows` instead.

chinpei215 commented on src/Shell/Helper/TableHelper.php:
```diff
@@ -106,6 +114,10 @@ public function output($rows)
             $this->_rowSeparator($widths);
         }
+    
+        if(empty($rows)) {
```
Same as above.

markstory commented:
Any way we could get tests for these scenarios ?

thinkingmedia commented:
What if it uses type hinting instead?

I only used empty because I didn't want to change the function signature, but it would be better with `output(array $row)` as that addresses the above comments.

@markstory arg.... okay.

ADmad commented:
Existing tests are also failing and there are CS errors.

ADmad commented:
@thinkingmedia ping

thinkingmedia commented:
I'll take a look at this today. Been sick.

Commit:
Merge remote-tracking branch 'upstream/master' into patch-4

Commit:
fixes unit tests, but I need to add some more tests

lorenzo commented on src/Shell/Helper/TableHelper.php:
```diff
@@ -78,6 +78,10 @@ protected function _rowSeparator($widths)
      */
     protected function _render($row, $widths, $options = [])
     {
+        if(!$row || !is_array($row)) {
```
`empty($rows)` or `count($rows)` would be shorter. Additionally, if rows is not an array it has the nice side effect that you will get an error :)

Commit:
give to phpcs what it wants

Commit:
use count()

thinkingmedia commented:
If everything passes then I'm finished.

lorenzo commented on src/Shell/Helper/TableHelper.php:
```diff
@@ -78,6 +78,10 @@ protected function _rowSeparator($widths)
      */
     protected function _render($row, $widths, $options = [])
     {
+        if (!is_array($row) || count($row) === 0) {
```
what's the need for checking if it is an array?

thinkingmedia replied:
there was no parameter hint in the original code, and I didn't want to change the function signature.

Now that I just wrote the above comment. I see that's silly. I'll add type hinting to the method and remove `is_array`

Commit:
removes is_array()
